consensus/tendermint/config.go

```diff
@@ -1,177 +1,42 @@
-// Copyright 2017 Monax Industries Limited
-//
-// Licensed under the Apache License, Version 2.0 (the "License");
-// you may not use this file except in compliance with the License.
-// You may obtain a copy of the License at
-//
-//    http://www.apache.org/licenses/LICENSE-2.0
-//
-// Unless required by applicable law or agreed to in writing, software
-// distributed under the License is distributed on an "AS IS" BASIS,
-// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-// See the License for the specific language governing permissions and
-// limitations under the License.
-
 package tendermint
 
 import (
 	"path"
-	"time"
-
-	"github.com/spf13/viper"
-	tendermintConfig "github.com/tendermint/go-config"
 
-	"github.com/hyperledger/burrow/config"
+	tm_config "github.com/tendermint/tendermint/config"
 )
 
-// NOTE [ben] Compiler check to ensure TendermintConfig successfully implements
-// tendermint/go-config/config.Config
-var _ tendermintConfig.Config = (*TendermintConfig)(nil)
-
-// Tendermint has a self-rolled configuration type defined
-// in tendermint/go-config but over an interface type, which is implemented
-// by default in tendermint/tendermint/config/tendermint.go
-// However, for burrow purposes we can choose different rules for how to load
-// the tendermint configuration and set the defaults.  Hence we re-implement
-// go-config.Config on a viper subtree of the loaded burrow configuration file.
-type TendermintConfig struct {
-	subTree *viper.Viper
-}
-
-func GetTendermintConfig(loadedConfig *viper.Viper) *TendermintConfig {
-	// ensure we make an explicit copy
-	subTree := new(viper.Viper)
-	*subTree = *loadedConfig
-
-	return &TendermintConfig{
-		subTree: subTree,
+// Burrow's view on Tendermint's config. Since we operate as a Tendermint harness not all configuration values
+// are applicable, we may not allow some values to specified, or we may not allow some to be set independently.
+// So this serves as a layer of indirection over Tendermint's real config that we derive from ours.
+type BurrowTendermintConfig struct {
+	Seeds                string
+	ListenAddress        string
+	Moniker              string
+	TendermintRoot       string
+}
+
+func DefaultBurrowTendermintConfig() *BurrowTendermintConfig {
+	tmDefaultConfig := tm_config.DefaultConfig()
+	return &BurrowTendermintConfig{
+		ListenAddress:        tmDefaultConfig.P2P.ListenAddress,
+		TendermintRoot:       ".burrow",
 	}
 }
 
-//------------------------------------------------------------------------------
-// Tendermint defaults
-
-//
-// Contract
-//
-
-func (tmintConfig *TendermintConfig) AssertTendermintDefaults(chainId, workDir,
```

Comment:
I like this more opinionated take on running tm. It will significantly declutter (and de-confuse) our configs.

Reply:
Good, I think we will need to expose more configurability, but rather than just checking every possible changing value I'd like us to expose a higher level interface to tuning that works with Burrow